src/helpers/collectReports.ts

```diff
@@ -16,7 +16,7 @@ export const collectReports = async ({
       reporters,
       async ({ name, instance }): Promise<Report[]> => {
         try {
-          return instance.getReports();
+          return await instance.getReports();
```

Comment:
> This is because even though the function is async and rejections in an async functions are converted to errors, the function above won't get a chance to catch the rejection thrown by getReports because that rejection happens after the the parent function returns. We have to await for getReports so that the parent function scope is kept and any errors thrown will be caught with the try { } catch { } block

That makes sense to me.

But does that mean the documentation of eslint is wrong? It says:

> Inside an async function, return await is useless

Reply:
It's only useless if you don't want to catch the error in the calling function.